pkg/apis/kubernetes/v1/cluster_types.go

```diff
@@ -0,0 +1,33 @@
+package v1
+
+import (
+	"context"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/apis/internal"
+)
+
+// anxcloud:object
+
+// Cluster represents a Kubernetes cluster
+// This resource does not support updates
+type Cluster struct {
+	internal.CommonMethods
+	State *internal.State `json:"state,omitempty"`
+
+	Identifier      string `json:"identifier,omitempty" anxcloud:"identifier"`
+	Name            string `json:"name,omitempty"`
+	Version         string `json:"version,omitempty"`
+	Location        string `json:"location_identifier,omitempty"`
+	NeedsServiceVMs *bool  `json:"needs_service_vms,omitempty"`
+	KubeConfig      string `json:"kubeconfig,omitempty"`
+}
+
+func (c *Cluster) getState() internal.State {
+	return *c.State
+}
+
+// AwaitCompletion blocks until the Cluster state is "OK"
+func (c *Cluster) AwaitCompletion(ctx context.Context, a api.API) error {
+	return awaitCompletion(ctx, a, c)
+}
```

Comment:
We don't need this as a method when allowing `awaitCompletion` to be called for any Object that implements `StateRetriever`